### PR TITLE
Deprecated functionality for v1.8

### LINKED ISSUE #5320: Docs Changes Needed from dbt-core Issue #9981
At a minimum, update body to include a link to the page on docs.getdbt.com requiring updates and what part(s) of the page you would like to see updated.

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Closes https://github.com/dbt-labs/docs.getdbt.com/issues/5320

Backported to v1.7 and v1.6

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/docs/dbt-versions/core-upgrade/01-upgrading-to-v1.8.md [hunk 1]
```diff
@@ -78,6 +78,20 @@ models:
...
 
+### Deprecated functionality
+
+The ability for installed packages to override built-in materializations without explicit opt-in from the user is being deprecated.
+
+- Overriding a built-in materialization from an installed package raises a deprecation warning.
+- Using a custom materialization from an installed package does not raise a deprecation warning.
+- Using a built-in materialization package override from the root project via a wrapping materialization is still supported. For example:
+
+  ```
```
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/core-upgrade/01-upgrading-to-v1.8.md]: should this be an sql code block? if it should, update all 3

```suggestion
  ```sql
```

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.6.md [hunk 2]
```diff
@@ -85,6 +85,20 @@ More consistency and flexibility around packages. Resources defined in a package
...
+
+- Overriding a built-in materialization from an installed package raises a deprecation warning.
+- Using a custom materialization from an installed package does not raise a deprecation warning.
+- Using a built-in materialization package override from the root project via a wrapping materialization is still supported. For example:
+
+  ```
+  {% materialization view, default %}
+  {{ return(my_cool_package.materialization_view_default()) }}
+  {% endmaterialization %}
+  ```
```
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.6.md:100]: wonder if this "deprecation notice" is worth capturing in a snippet/partial/reusable

**Review by nghi-ly — APPROVED** [review]
looks good! approving with just one sugg to consider

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.6.md (hunk 2 above)
Comment (matthewshaver) [reply website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.6.md:100]: It's only going in the migration guides. When we actually deprecate it in 1.9 I'll see how extensive it is. Snippet might be a good call then.

**Pull request merged**